packages/core/test/ServerTest.ts

```diff
@@ -116,13 +155,161 @@ class WoTServerTest {
+        expect(await thing.readProperty("number")).to.equal(2);
+        expect(await thing.readProperty("number")).to.equal(3);
+    }
+
+    @test async "should be able to add a property, read it by setting write handler double"() {
+        let thing: WoT.ExposedThing = WoTServerTest.WoT.produce({ name: "otherthingWrite" });
+        let initp: WoT.ThingProperty = {
+            name: "number",
+            writable: true,
+            schema: `{ "type": "number" }`
+        };
+        thing.addProperty(initp);
+        let initp2: WoT.ThingProperty = {
+            name: "number2",
+            writable: true,
+            schema: `{ "type": "number" }`
+        };
+        thing.addProperty(initp2);
+        thing.setPropertyWriteHandler(
+            initp.name,
+            (value: any) => {
+                return new Promise((resolve, reject) => {
+                    thing.writeProperty(initp2.name, value*2);
+                    resolve(value);
+                });
+            }
+        );
+
+        expect(await thing.writeProperty("number", 12)).to.equal(12);
+        expect(await thing.readProperty("number2")).to.equal(24);
+
+        thing.setPropertyWriteHandler(
+            initp.name,
+            function(value: any) {
+                return new Promise((resolve, reject) => {
+                    thing.writeProperty(initp2.name, value*2);
+                    resolve(value);
+                });
+            }
+        );
+
+        expect(await thing.writeProperty("number", 13)).to.equal(13);
+        expect(await thing.readProperty("number2")).to.equal(26);
+    }
+
+    @test async "should be able to add a property, read it by setting write handler with reading old value"() {
+        let thing: WoT.ExposedThing = WoTServerTest.WoT.produce({ name: "otherthingReadWrite" });
+        let initp: WoT.ThingProperty = {
+            name: "number",
+            writable: true,
+            schema: `{ "type": "number" }`,
+            value : 2
+        };
+        thing.addProperty(initp);
+        // set handler that writes newValue as oldValue+request
+        thing.setPropertyWriteHandler(
+            initp.name,
+            (value: any) => {
+                return new Promise((resolve, reject) => {
+                    thing.readProperty(initp.name).then(
+                         (oldValue) => {
+                            resolve(oldValue + value);
+                         }
+                    );
+                });
+            }
+        );
+
+        expect(await thing.writeProperty("number", 1)).to.equal(3); // 2 + 1 = 3
+        expect(await thing.readProperty("number")).to.equal(3);
+
+        expect(await thing.writeProperty("number", 2)).to.equal(5); // 3 + 2 = 5
+        expect(await thing.readProperty("number")).to.equal(5);
+    }
+
+
     // FIXME What listeners?!
     @test.skip "should be able to add a property, assign it via listener and read it locally"() {
```

Comment:
What is this test for?
If also no clue, please just delete.
There are no Property listeners; maybe this is from pre-Obersable times.